lib/utils/tty.js

```diff
@@ -5,9 +5,27 @@
 var _ = require("lodash");
 var readline = require("readline");
 var cliWidth = require("cli-width");
+var utils = require("./utils");
 
 var tty = module.exports;
 
+/**
+ * Get the number of lines the current line occupies
+ * @return {Number}  The number of lines
+ */
+
+tty.lines = function() {
+  var value = this.rl.line || this.rl.history[0];
+  var lineLength = utils.getUnicodeStringLength( this.rl._prompt + value );
+  var result = lineLength / cliWidth();
```

Comment:
Seems like a potential workaround.

Another is to manually handle line returns so we stay consistent. But that might just bring new kinds of issues.